source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -0,0 +1,372 @@
+/*
+  Copyright (C) 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/composite_visco_plastic.h>
+#include <aspect/material_model/utilities.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      // The following function is not yet used by ASPECT, so
```

Comment:
At some point a comment should be added here (or elsewhere) that even a specific flow law mechanism is not used (ex: Peierls), there will still be a strain rate value output for that mechanism as the default flow law viscosity are the user-defined maximum viscosities.

Reply:
That isn't quite true. If a specific flow law mechanism is not used the strain rate for that mechanism will be exactly zero. To avoid unnecessary complexity, I've fixed the length and order of the outputs, so a vector may contain multiple outputs. I've extended the comment to make this clear.

Reply:
Thanks for extending the comment and really slick design!